### DISCUSSION TITLE
pnpm support: What is meant by unsupported ?

### DISCUSSION THREAD

**Discussion by cassepipe** [skeletonlabs/skeleton#1165] [Q&A] [answered]
Sorry if my question is silly, I am a total beginner in webdev but the FAQ says :

```We support [NPM (Node Package Manager)](https://docs.npmjs.com/about-npm). PNPM and Yarn are currently only supported for the Skeleton CLI.```

From what I had understood, `pnpm` only creates hardlinks so that dependencies are effectively reused across projects and I am personnally using it as a drop-in replacement for `npm`.

So what does it mean for `pnpm` to be (un)supported ?

**Comment by Sarenor** [comment, marked as answer]
We haven't found issues with ``pnpm`` yet and some of our contributors use it for Skeleton development as well.
Afaik unsupported in this case means that if you do run into issues, we'll probably take a look and try to fix it, but not with the same priority and urgency as if it is an issue that can be replicated with npm.

If we can't fix it or don't know what's causing it, it might also remain open and will, again, not be treaded with priority.

Although I use and like pnpm myself, so I'll probably spent some time with it if I find said time ;)

**Comment by endigo9740** [comment]
What Sarenor said, but to add the "why" here - we're still a small team so we keep the scope limited for now. Now if someone wanted to come in and be the point person on PNPM, Yarn, or similar we would gladly extend support. But for right now no one has stepped into that role.

**Reply by cassepipe** [reply]
I just thought unsupported meant it didn't really work/ would be broken for technical ${reasons}.

Of course, you have limited time and ressources. Thanks for providing free and open source (and useful) software <3

**Reply by endigo9740** [reply]
Unsupported here means two things:

1. It may not operate as intended. There may be extra steps involved which are not documented.
2. If it doesn't work as intended we may be limited in scope for how we can help.

So it's both sides really. Essentially there's no guarantee it will (or won't) work. But again, it's our understanding that PNPM specifically is working fine as is. No clue about Yarn though! It's untested as far as I'm aware.

We try to state this clearly so folks don't come yelling at us because it's not meeting expectations. This happens more than you might expect :)
